core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -778,41 +692,15 @@ private[spark] class ExternalSorter[K, V, C](
 
     context.taskMetrics.incMemoryBytesSpilled(memoryBytesSpilled)
     context.taskMetrics.incDiskBytesSpilled(diskBytesSpilled)
-    context.taskMetrics.shuffleWriteMetrics.filter(_ => bypassMergeSort).foreach { m =>
```

Comment:
To quickly clarify, this `filter` is equivalent to an `if(bypassMergeSort)` check.  I think that I haven't broken any of the metrics here for the non-bypass-merge-sort case, but it's a little tricky to tell because `ExternalSorterSuite` doesn't seem to have any tests that check the ShuffleWriteMetrics or task metrics.  I can look into adding these tests but I'd like to do it in a separate followup PR.